app/post_results.go

```diff
@@ -259,13 +255,85 @@ func extractAndVerifyCertForPayload(ctx context.Context,
 	}
 
 	cert := certs[0]
+
 	// Verify that cert isn't expired.
-	if err = cosign.CheckExpiry(cert, time.Unix(*entry.IntegratedTime, 0)); err != nil {
-		return nil, fmt.Errorf("error during cosign.CheckExpiry: %w", err)
+	integratedTime := time.Unix(entry.IntegratedTime, 0)
+	if cert.NotAfter.Before(integratedTime) {
+		return nil, fmt.Errorf("certificate expired before signatures were entered in log: %s is before %s",
+			cert.NotAfter, integratedTime)
+	}
+	if cert.NotBefore.After(integratedTime) {
+		return nil, fmt.Errorf("certificate was issued after signatures were entered in log: %s is after %s",
+			cert.NotAfter, integratedTime)
 	}
 	return cert, nil
 }
 
+func getUUIDsByPayload(ctx context.Context, payload []byte) ([]string, error) {
+	payloadSHA := sha256.Sum256(payload)
+	rekorPayload := struct {
+		Hash string `json:"hash"`
+	}{
+		Hash: fmt.Sprintf("sha256:%s", hex.EncodeToString(payloadSHA[:])),
+	}
+	jsonPayload, err := json.Marshal(rekorPayload)
+	if err != nil {
+		return nil, fmt.Errorf("marshaling json payload: %w", err)
+	}
+
+	rekorReq, err := http.NewRequestWithContext(ctx,
+		http.MethodPost,
+		"https://rekor.sigstore.dev/api/v1/index/retrieve",
```

Comment:
We are using `prod`. Maybe use the `staging`  https://rekor.sigstage.dev.  Here is something that is being used https://github.com/slsa-framework/slsa-github-generator/pull/433

Reply:
Why would we use `staging`? Isn't `prod` the stable version?

Reply:
@laurentsimon/ @asraa can add more info on this. It was recommended that we use staging as prod is having some issues.